GameEngine/Engine.cs

```diff
@@ -8,7 +8,8 @@ namespace GameEngine
 {
     public class Engine
     {
-        string NewPlayerName;
+        private GameState CurrentGameState { get; set; }
```

Comment:
If CurrentGameState is not static that means we can have multiple current game states ?